File: postIt-frontend/lib/model/member.dart
class Member {
  String? username;
  String? password;
  String? nickname;
  String? email;
  String? loginType;

  Map<String, dynamic> toMap() {
    return {
      'username': this.username,
      'password': this.password,
      'nickname': this.nickname,
      'email': this.email,
      'loginType': this.loginType,
    };
  }

  factory Member.fromMap(Map<String, dynamic> map) {
    return Member(
      username: map['username'] as String,
      password: map['password'] as String,
      nickname: map['nickname'] as String,
      email: map['email'] as String,
      loginType: map['loginType'] as String,
    );
  }

  Member({
    this.username,
    this.password,
    this.nickname,
    this.email,
    this.loginType,
  });
}

File: postIt-backend/src/main/java/com/example/postItBackend/domain/auth/model/Member.java
package com.example.postItBackend.domain.auth.model;

import com.example.postItBackend.domain.BaseEntity;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "member", indexes = @Index(name = "idx_username", columnList = "username", unique = true))
@Getter
@NoArgsConstructor
public class Member extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private Long id;

    @Column(nullable = false, unique = true)
    private String username;

    @Column(nullable = false)
    private String nickname;

    @Column(nullable = false)
    private String email;

    @Column(nullable = true)
    private String password;

    @Column(nullable = false)
    private String loginType;

    @Column(nullable = true)
    private String providerId;

    @Builder
    public Member(Long id, String username, String nickname, String email, String password, String loginType) {
        this.id = id;
        this.username = username;
        this.nickname = nickname;
        this.email = email;
        this.password = password;
        this.loginType = loginType;
    }
}
